### PR TITLE
Component/checkbox

### PR DESCRIPTION
**Pull request by Rodrigo-Schar**
added checkbox component

### CODE REVIEW

**Comment by Rodrigo-Schar** [comment]
image

**Comment by Rodrigo-Schar** [comment]
image

File: src/components/Checkbox.tsx [hunk 1]
```diff
@@ -1,15 +1,52 @@
...
+  const [checked, setChecked] = useState(false);
+  const styles = useStyles(checked);
 
-// TODO
-const Checkbox = () => {
   return (
-    <View>
-      <Text>Checkbox</Text>
-    </View>
+    <TouchableWithoutFeedback onPress={onPress}>
```
[the remark this answers is not in the document]
Comment (Rodrigo-Schar) [reply src/components/Checkbox.tsx:18]: i think is done check it

File: src/components/Checkbox.tsx [hunk 2]
```diff
@@ -1,15 +1,52 @@
...
-import React from 'react'
+import { StyleSheet, Text, View, TouchableWithoutFeedback, TouchableWithoutFeedbackProps } from 'react-native'
+import { useContext, useState } from 'react'
+import { ColorContext } from '../shared/ColorContext';
+
+type CheckboxProps = {
+  text: string;
+}
+
+const Checkbox = ({text, onPress}: CheckboxProps & TouchableWithoutFeedbackProps) => {
```
[the remark this answers is not in the document]
Comment (Rodrigo-Schar) [reply src/components/Checkbox.tsx]: I'm not understanding can you explain it better?

File: src/components/Checkbox.tsx [hunk 3]
```diff
@@ -1,15 +1,56 @@
...
+  text: string;
+}
+
+const Checkbox = ({text}: CheckboxProps & TouchableWithoutFeedbackProps) => {
+  const [checked, setChecked] = useState(false);
+  const styles = useStyles(checked);
+
+  const onPress = () => {
+    checked == true ? setChecked(false) : setChecked(true)
+  };
```
Comment (gaguayovelez) [inline src/components/Checkbox.tsx:15]:
```ts
setChecked(!checked);
```

File: src/components/Checkbox.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaguayovelez) [reply src/components/Checkbox.tsx:18]: @Rodrigo-Schar you need expose the following properties:

```ts

type CheckboxProps = {
  text: string;
  value: boolean,
  onChange: (value: boolean) => void
}
```

**Review by gaguayovelez — CHANGES_REQUESTED** [review]

**Pull request merged**